backend/audit/utils.py

```diff
@@ -72,6 +73,22 @@ def remove_extra_fields(general_information_data):
             general_information_data.pop("audit_period_other_months", None)
         return general_information_data
 
+    @staticmethod
+    def check_leap_year(end_date):
```

Comment:
*Dates. I hate dates.* (With apologies to Indiana Jones.)

Is there a way we can use the date library to 

1. Do the calculation and
2. Put a test or two in place that asserts the things that you're doing in your code?

My concern is this: 

*Are there other cases you haven't considered?*

In this regard, if there's a way we can say

```
if calendar.isleap(...):
  return calendar.subtract_one_year(...)
```

so that the question of 365 vs. 366 is handled by someone else.

It would also be good for us to identify examples where this hit us, and make sure we have some tests that make certain that, when we use this function in the future, it really does the right thing on the edges we're worried about.

If that doesn't make sense, feel free to ask questions.

Reply:
@sambodeme has found one of the tickets this morning related to this issue [here](https://fac-gov.zendesk.com/agent/tickets/4337). However, it is not fully clear to me that the audits involved are an issue _because of_ the leap year. If there are other helpdesk tickets related to this, getting some sample dates and expected outcomes would be very useful.

I did find a library **dateutil.relativedelta** which does seem to take into account the current year like you asked. From what I've read about it that means leap years too. That allows us to do something akin to:
```
start_date = end_date - relativedelta(years=1)
```

Whereas what we had before was `timedelta(days=365)`. But I would rather not continue to tweak on this until there are clear start/end dates from the helpdesk that I can experiment these changes on.

Reply:
I think pausing tweaking is reasonable; good call. We have a breadcrumb trail on your thinking here, which is good.

Reply:
I think we are good to proceed. Speaking with Hassan and Dan, the new changes I am posting to this PR resolves the original issue.

https://app.fac.gov/dissemination/summary/2022-06-CENSUS-0000015322
The expected fiscal range is `7/1/2021` -> `6/30/2022`. I have adjusted one of the start dates from the `census_historical_migration` tests to be 6/30/2022. I am still unsure about the leap year piece, however, after running the above dates (from my PR description) against the new `dateutil` library, I am getting the output I expect.